sdk/eventgrid/azure-messaging-eventgrid/migration-guide.md

```diff
@@ -0,0 +1,141 @@
+for those interested.
+
+### Cross Service SDK improvements
+
+The modern Event Hubs client library also provides the ability to share in some of the cross-service improvements made to the Azure development experience, such as
+
+- Using the new azure-identity library to share a single authentication approach between clients.
+- A unified logging and diagnostics pipeline offering a common view of the activities across each of the client libraries.
+- A unified asynchronous programming model using [Project Reactor][project-reactor].
+- A unified method of creating clients via client builders to interact with Azure services.
+
+## Important changes
+
+### Group id, artifact id, and package names
+
+Group ids, artifact ids, and package names for the modern Azure client libraries for Java have changed. They follow the [Java SDK naming guidelines][GuidelinesJavaDesign]. Each will have the group id `com.azure`, an artifact id following the pattern `azure-[area]-[service]`, and the root package name `com.azure.[area].[Service]`. The legacy clients have a group id `com.microsoft.azure` and their package names followed the pattern `com.microsoft. Azure.[service]`. This provides a quick and accessible means to help understand, at a glance, whether you are using modern or legacy clients.
+
+In EventGrid, the modern client libraries have packages and namespaces that begin with `com.azure.messaging.eventgrid` and were released starting with version 4. The legacy client libraries have package names starting with `com.microsoft.azure.eventgrid` and a version of 1.x.x or below.
+
+#### Instantiating clients
+
+In v1, the `EventGridClient` was instantiated via the `EventGridClientImpl` constructor. The client contained both sync and async methods.
+The client can send only `EventGridEvent` to an EventGrid topic or domain.
+```java
+String endpoint = "<endpont of your event grid topic/domain that accepts EventGridEvent schema>";
+TopicCredentials topicCredentials = new TopicCredentials(key);
+EventGridClient client = new EventGridClientImpl(topicCredentials);
+```
+
+In v4, the creation of the client is done through the [EventGridPublisherClientBuilder][EventGridPublisherClientBuilder]. The sync and async operations are separated to [EventGridPublisherClient] and [EventGridPublisherAsyncClient].
+```java
+EventGridPublisherClient<EventGridEvent> eventGridEventClient = new EventGridPublisherClientBuilder()
+    .endpoint("<endpont of your event grid topic/domain that accepts EventGridEvent schema>")
+    .credential(new AzureKeyCredential("<key for the endpoint>"))
+    .buildEventGridEventPublisherClient();
+```
+Aside from [EventGridEvent][EventGridEvent], the generic client can be instantiated to also send `CloudEvent`, or custom events to an EventGrid topic or domain.
+Refer to the [README][README] for more information about sending `CloudEvent` and custom events.
+
+#### Send events to an Event Grid Topic
```

Comment:
Other topics this Migration guide should include:
- sending custom event
- receiving custom event
- sending to domain

Reply:
Added "sending to domain". The old SDK doesn't have sending to custom event / cloud event so this migration guide may not need it.